Don't consider 'fill' a presentation attribute on animation elements

Ahmad-S792 opened the pull request:
#### a0c45ccc4d6a65ce7f2df1f684cfe0abc03f07cd
<pre>
Don&apos;t consider &apos;fill&apos; a presentation attribute on animation elements

<a href="https://bugs.webkit.org/show_bug.cgi?id=274796">https://bugs.webkit.org/show_bug.cgi?id=274796</a>
<a href="https://rdar.apple.com/problem/128896937">rdar://problem/128896937</a>

Reviewed by Simon Fraser.

Merge: <a href="https://chromium-review.googlesource.com/c/chromium/src/+/2080237">https://chromium-review.googlesource.com/c/chromium/src/+/2080237</a>

This patch aligns WebKit with Gecko / Firefox,
Blink / Chromium and web-specification [1]:

[1] <a href="https://svgwg.org/svg2-draft/styling.html#PresentationAttributes">https://svgwg.org/svg2-draft/styling.html#PresentationAttributes</a>

Elements that support the presentation attribute:

&quot;Any element in the SVG namespace except for animation elements,
which have a different ‘fill’ attribute.&quot;

* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::hasPresentationalHintsForAttribute const):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/presentation-attributes-special-cases-expected.txt: Rebaselined

Canonical link: <a href="https://commits.webkit.org/282100@main">https://commits.webkit.org/282100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e9e0627eba113435e0a147d29c7999d543763169

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62091 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/41445 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/14683 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66071 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/12636 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/64210 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49131 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/12976 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50005 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/8730 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65160 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/38466 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/53773 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/30836 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35110 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11009 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/11567 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/56905 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/11313 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/67799 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6034 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11076 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/57383 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6060 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/53722 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/57633 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13797 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/4946 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/37245 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/38329 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/39425 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38074 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/svg/animation/SVGSMILElement.cpp:
```diff
@@ -213,6 +213,14 @@ void SVGSMILElement::buildPendingResource()
 
+bool SVGSMILElement::hasPresentationalHintsForAttribute(const QualifiedName& name) const
+{
+    // Don't map 'fill' to the 'fill' property for animation elements.
```
This could use a spec link; I can't tell from the comment if it's for spec or performance or other reasons.

Ahmad-S792 replied:
Sure - will add 'spec' link (similar to commit message) after EWS run before merging. :-)

Commit:
Don't consider 'fill' a presentation attribute on animation elements

https://bugs.webkit.org/show_bug.cgi?id=274796
rdar://problem/128896937

Reviewed by Simon Fraser.

Merge: https://chromium-review.googlesource.com/c/chromium/src/+/2080237

This patch aligns WebKit with Gecko / Firefox,
Blink / Chromium and web-specification [1]:

[1] https://svgwg.org/svg2-draft/styling.html#PresentationAttributes

Elements that support the presentation attribute:

"Any element in the SVG namespace except for animation elements,
which have a different ‘fill’ attribute."

* Source/WebCore/svg/animation/SVGSMILElement.cpp:
(WebCore::SVGSMILElement::hasPresentationalHintsForAttribute const):
* Source/WebCore/svg/animation/SVGSMILElement.h:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/presentation-attributes-special-cases-expected.txt: Rebaselined

Canonical link: https://commits.webkit.org/282100@main
